tribe-ext-disallow-past-starting-dates.php

```diff
@@ -0,0 +1,517 @@
+	 */
+	class Tribe__Extension__Custom_Datepicker_Start_Date extends Tribe__Extension {
+		/**
+		 * The script's handle.
+		 */
+		public $handle = 'tribe-ext-custom-datepicker-start-date';
+
+		/**
+		 * The array of keys/values to send to the script.
+		 *
+		 * @see wp_localize_script()
+		 *
+		 * @var array
+		 */
+		private $script_vars = array();
+
+		/**
+		 * The existing start date's midnight timestamp.
+		 *
+		 * This will remain NULL if this is a new (non-existing) event or if its
+		 * start date is TODAY.
+		 *
+		 * @see Tribe__Extension__Custom_Datepicker_Start_Date::get_min_date()
+		 *
+		 * @var null|int
+		 */
+		private $timestamp_existing_start_date = null;
+
+		/**
+		 * The timestamp of whichever is greater between Today and an existing
+		 * event's start time, if applicable.
+		 *
+		 * This will remain NULL if this is a new (non-existing) event or if its
+		 * start date is TODAY.
+		 *
+		 * @see Tribe__Extension__Custom_Datepicker_Start_Date::get_min_date()
+		 *
+		 * @var null|int
+		 */
+		private $timestamp_max_today_or_existing = null;
+
+		/**
+		 * The script's handle with underscores instead of hyphens.
+		 *
+		 * Used for building filter and action hook names and CSS class name.
+		 *
+		 * @return string
+		 */
+		public function get_handle_underscores() {
+			return str_replace( '-', '_', $this->handle );
+		}
+
+		/**
+		 * The input's CSS error class.
+		 *
+		 * @return string
+		 */
+		private function get_error_css_class() {
+			return $this->get_handle_underscores() . '_error';
+		}
+
+		/**
+		 * The WordPress capability required to be able to pick any date.
+		 *
+		 * @return string
+		 */
+		private function get_cap_allowed_any_date() {
+			/**
+			 * The capability required to have this script not load.
+			 *
+			 * @param string $capability The minimum capability to be allowed to
+			 *                           choose any start date.
+			 *
+			 * @return string
+			 */
+			return (string) apply_filters( $this->get_handle_underscores() . '_cap_allowed_any_start_date', 'manage_options' );
+
+		}
+
+		/**
+		 * Set the minimum required version of The Events Calendar
+		 * and this extension's URL.
+		 */
+		public function construct() {
+			$this->add_required_plugin( 'Tribe__Events__Main' );
+			$this->set_url( 'https://theeventscalendar.com/extensions/custom-datepicker-start-date/' );
+		}
+
+		/**
+		 * Extension initialization and hooks.
+		 */
+		public function init() {
+			// Load plugin textdomain
+			load_plugin_textdomain( $this->handle, false, basename( dirname( __FILE__ ) ) . '/languages/' );
+
+			// Requires PHP 5.3+ to use DateTime::setTimestamp() and the DateInterval class.
+			if ( version_compare( PHP_VERSION, '5.3', '<' ) ) {
+				$message = '<p>' . $this->get_name();
+
+				$message .= __( ' requires PHP 5.3 or newer to work. Please contact your website host and inquire about updating PHP.', 'tribe-ext-custom-datepicker-start-date' );
```

Comment:
We should probably use `sprintf()` here. By concatenating the name always comes first in the output, but in _language XYZ_ that might not make sense, whereas a `%1$s` placeholder can be moved around by the translator as appropriate.

Reply:
done